data/src/environment.rs

```diff
@@ -43,3 +35,49 @@ fn portable_dir() -> Option<PathBuf> {
         .is_file()
         .then(|| dir.to_path_buf())
 }
+
+fn platform_specific_config_dir() -> PathBuf {
+    #[cfg(target_os = "macos")]
+    {
+        xdg_config_dir().unwrap_or_else(|| {
+            dirs_next::config_dir()
+                .expect("expected valid config dir")
+                .join("halloy")
+        })
+    }
+    #[cfg(not(target_os = "macos"))]
+    {
+        dirs_next::config_dir()
+            .expect("expected valid config dir")
+            .join("halloy")
+    }
+}
+
+fn platform_specific_data_dir() -> PathBuf {
+    #[cfg(target_os = "macos")]
+    {
+        xdg_config_dir().unwrap_or_else(|| {
```

Comment:
As we discussed, let's not change `data_dir` behavior. We especially DO NOT want to use XDG_CONFIG_HOME to store data files.